File: src/main/java/com/now/nowbot/entity/ServiceCallLite.java
package com.now.nowbot.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "service_call")
public class ServiceCallLite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String service;
    Long   time;

    @Column(name = "ctime")
    LocalDateTime createTime;

    public ServiceCallLite() {
        createTime = LocalDateTime.now();
    }

    public ServiceCallLite(String service, Long time) {
        this.service = service;
        this.time = time;
        createTime = LocalDateTime.now();
    }


    public interface ServiceCallResult {
        String getService();

        Integer getSize();

        Double getAvgTime();

        Long getMinTime();

        Long getMaxTime();
    }


    public interface ServiceCallResult$80 {
        String getService();

        Long getData();
    }
}

File: src/main/java/com/now/nowbot/entity/ServiceCallLite.kt
package com.now.nowbot.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity @Table(name = "service_call")
class ServiceCallLite {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(name = "service")
    var name: String = ""

    @Column(name = "time")
    var duration: Long = 0L

    @Column(name = "ctime")
    var createTime: LocalDateTime = LocalDateTime.now()

    companion object {
        fun ServiceCallLite.toStatistic(): ServiceCallStatistic {
            return ServiceCallStatistic(
                id = id,
                name = name,
                duration = duration,
                createTime = createTime,
            )
        }
    }
}
